client/my-sites/plugins/main.jsx

```diff
@@ -268,29 +268,13 @@ const PluginsMain = React.createClass( {
 	},
 
 	renderPluginsContent() {
-		const plugins = this.state.plugins || [];
-		const { selectedSite } = this.props;
+		const { plugins = [] } = this.state;
+		const { filter, search, selectedSite } = this.props;
 
-		if ( isEmpty( plugins ) && ! this.isFetchingPlugins() ) {
-			if ( this.props.search ) {
-				const searchTitle = this.props.translate( 'Suggested plugins for: %(searchQuery)s', {
-					textOnly: true,
-					args: {
-						searchQuery: this.props.search
-					}
-				} );
-
-				return <PluginsBrowser
-						hideSearchForm
-						site={ selectedSite ? selectedSite.slug : null }
-						path={ this.context.path }
-						sites={ this.props.sites }
-						search={ this.props.search }
-						store={ this.context.store }
-						searchTitle={ searchTitle }
-					/>;
-			}
+		const showInstalledPluginList = ! isEmpty( plugins ) || this.isFetchingPlugins();
+		const showSuggestedPluginsList = filter === 'all' || ( ! showInstalledPluginList && search );
```

Comment:
Possible that `showInstalledPluginList` and `showSuggestedPluginsList` could be props created in `connect`? Would involve making `isFetchingPlugins` a prop as well. May not be worth changing things around.